### PR TITLE
feat: support execution context in workflowrun

### PR DESCRIPTION
**Pull request by cd1989**
**What this PR does / why we need it**:

Support execution context to give namespace and PVC in WorkflowRun.

Note: Multi-cluster not implemented in this PR.

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @zhujian7 @supereagle 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/cyclone/v1alpha1/workflow_run.go [hunk 1]
```diff
@@ -29,24 +29,37 @@ type WorkflowRunSpec struct {
...
 // ParameterConfig configures parameters of a resource or a stage.
 type ParameterConfig struct {
 	// Whose parameters to configure
 	Name string `json:"name"`
-	// Parameters
+	// Parameters ...
 	Parameters []ParameterItem `json:"parameters"`
 }
 
+// ExecutionContext defines execution context of a workflow, namespace, pvc
```
Comment (zhujian7) [inline pkg/apis/cyclone/v1alpha1/workflow_run.go]: could you please write `namespace, pvc cluster would be specified` in a new line to separate `... a workflow` with `namespace, pvc ...`

File: pkg/workflow/workflowrun/pod.go [hunk 2]
```diff
@@ -75,22 +76,14 @@ func (m *PodBuilder) Prepare() error {
...
-		Namespace: m.wfr.Namespace,
+		Namespace: m.executionContext.Namespace,
 		Labels: map[string]string{
 			common.WorkflowLabelName: "true",
 		},
 		Annotations: map[string]string{
 			common.WorkflowRunAnnotationName: m.wfr.Name,
 			common.StageAnnotationName:       m.stage,
 		},
-		OwnerReferences: []metav1.OwnerReference{
```
Comment (zhujian7) [inline pkg/workflow/workflowrun/pod.go]: `OwnerReferences` deleted, pod will still running even if the workflowrun deleted? Should we file an issue to track it?

Comment (cd1989) [reply pkg/workflow/workflowrun/pod.go]: Pod would be deleted by controller as before when it the pod terminated.

File: Makefile [hunk 3]
```diff
@@ -51,6 +51,9 @@ SCENE ?= cicd
 # PVC used to run workflow
 PVC ?=
 
+# Namespace where workflow would run, the above PVC should belong to this namespace
+EXEC_NAMESPACE ?= default
```
Comment (supereagle) [inline Makefile]: Install and run in `cyclone` namespace in default?

File: pkg/workflow/controller/config.go [hunk 4]
```diff
@@ -52,12 +52,8 @@ type WorkflowControllerConfig struct {
 	Limits LimitsConfig `json:"limits"`
 	// ResourceRequirements is default resource requirements for containers in stage Pod
 	ResourceRequirements corev1.ResourceRequirements `json:"default_resource_quota"`
```
Comment (supereagle) [inline pkg/workflow/controller/config.go]: Also camelCase here.

File: pkg/workflow/controller/config.go [hunk 5]
```diff
@@ -52,12 +52,8 @@ type WorkflowControllerConfig struct {
...
-	// PVC used to transfer artifacts in WorkflowRun, and also to help share resources
-	// among stages within WorkflowRun. If no PVC is given here, input resources won't be
-	// shared among stages, but need to be pulled every time it's needed. And also if no
-	// PVC given, artifacts are not supported.
-	// TODO(ChenDe): Remove it when Cyclone can manage PVC for namespaces.
-	PVC string `json:"pvc"`
+	// ExecutionContext defines default namespace and pvc used to run workflow.
+	ExecutionContext ExecutionContext `json:"execution_context"`
```
Comment (supereagle) [inline pkg/workflow/controller/config.go]: camelCase for json?

File: pkg/workflow/controller/config.go [hunk 6]
```diff
@@ -70,6 +66,19 @@ type LoggingConfig struct {
...
+type ExecutionContext struct {
+	// Namespace is namespace where to run workflow.
+	Namespace string `json:"namespace"`
+	// PVC is pvc used to run workflow. It's used to transfer artifacts in WorkflowRun, and
+	// also to help share resources among stages within WorkflowRun. If no PVC is given here,
+	// input resources won't be shared among stages, but need to be pulled every time it's needed.
+	// And also if no PVC given, artifacts are not supported.
+	PVC string `json:"pvc"`
+	// ServiceAccount is the service account applied to the pod runed
+	ServiceAccount string `json:"service_account"`
```
Comment (supereagle) [inline pkg/workflow/controller/config.go]: camelCase for json?

**Pull request merged**